### PR TITLE
[Web] Unified Resources: Add skeleton loader to list view

### PR DESCRIPTION
**Pull request by rudream**
## Purpose

Based on https://github.com/gravitational/teleport/pull/34203

Adds a skeleton loader for the loading state in the unified resources list view.

Note: I also made a tweak to the colors we use in `ShimmerBox.tsx`, currently it looks great in dark theme but in light theme I thought the contrast was too low and it was hard to see the effect (comparison below).

Will address conflicts and merge once https://github.com/gravitational/teleport/pull/34384 is merged.

## Demo

https://github.com/gravitational/teleport/assets/56373201/09343146-06d5-44dd-9b01-95d44fd20594

https://github.com/gravitational/teleport/assets/56373201/70ba54d8-8dab-4f0e-8a8f-43dd95081c4c

### Old `ShimmerBox` in light theme:

https://github.com/gravitational/teleport/assets/56373201/46395e1b-ccf4-4837-b139-1e1fcc5ba4b3

### CODE REVIEW

**Review by avatus — COMMENTED** [review]
Check [this recent PR](https://github.com/gravitational/teleport/pull/34384#pullrequestreview-1722869035) that changes the way we `setTimeout` for the loading skeletons. I think everything in your PR is fine, but we should update it when this other PR goes in to follow the same

File: web/packages/shared/components/UnifiedResources/CardsView/LoadingCard.tsx
Comment (avatus) [inline web/packages/shared/components/UnifiedResources/CardsView/LoadingCard.tsx:40]: I think we don't need the timeout here anymore once @gzdunek changes go through as he updated the way the card loader works. 

ListViewSkeleton looks great

premptive approve once the other PR goes in and conflicts resolved here

**Review by avatus — APPROVED** [review]

File: web/packages/shared/components/UnifiedResources/ListView/LoadingListItem.tsx
Comment (gzdunek) [inline web/packages/shared/components/UnifiedResources/ListView/LoadingListItem.tsx:51]: After my PR is merged we will be able to export only this component and pass it to `<LoadingSkeleton />`.

File: web/packages/shared/components/UnifiedResources/types.ts
Comment (gzdunek) [inline web/packages/shared/components/UnifiedResources/types.ts:191]: This won't be needed at all.

File: web/packages/shared/components/UnifiedResources/CardsView/LoadingCard.tsx
Comment (gzdunek) [inline web/packages/shared/components/UnifiedResources/CardsView/LoadingCard.tsx:1]: In my PR I added `LoadingCard` too, we will have to move it from `UnifiedResources` to here.

File: web/packages/shared/components/UnifiedResources/ListView/ListView.tsx
Comment (gzdunek) [inline web/packages/shared/components/UnifiedResources/ListView/ListView.tsx:56]: Copy & paste comment? :)

**Review by gzdunek — COMMENTED** [review]
Looks good, I will take another look today after we resolve conflicts.

**Review by gzdunek — APPROVED** [review]

**Comment by rudream** [comment]
Added `e` ref update for https://github.com/gravitational/teleport.e/pull/2619 to this PR as well.

**Pull request merged**
